framework/contracts/account/manager/src/queries.rs

```diff
@@ -94,12 +100,34 @@ pub fn handle_sub_accounts_query(
 }
 
 /// RawQuery the version of an enabled module
-pub fn query_module_cw2(deps: &Deps, module_addr: Addr) -> StdResult<ContractVersion> {
-    let req = QueryRequest::Wasm(WasmQuery::Raw {
-        contract_addr: module_addr.into(),
-        key: CONTRACT.as_slice().into(),
-    });
-    deps.querier.query::<ContractVersion>(&req)
+pub fn query_module_version(
+    deps: Deps,
+    module_addr: Addr,
+    version_control: &VersionControlContract,
+) -> StdResult<ContractVersion> {
+    let module_registry = version_control.module_registry(deps);
+
+    // Currently manager doesn't save ModuleReference,
+    // so we need to check for standalone first
+    let code_id = deps
```

Comment:
Can we change this logic to only perform the code-id + standalone query if the cw2 query fails OR if the query succeeds but the cw2 name doesn't adhere to our formatting standards? Should be more efficient

Reply:
Implemented, thanks